kernel/base/src/main/java/com/twosigma/beakerx/AutotranslationServiceImpl.java

```diff
@@ -0,0 +1,143 @@
+  public static final String LOCALHOST = "http://localhost:";
+  public static final String AUTOTRANSLTION = "/autotransltion/";
+  private final String contextAsString;
+  private final AutotranslationContext context;
+
+  public static AutotranslationService createAsSubkernel(String configuration) {
+    return new AutotranslationServiceImpl(configuration);
+  }
+
+  public static AutotranslationService createAsMainKernel(String id) {
+    Map<String, String> context = new HashMap<>();
+    context.put("contextId", id);
+    context.put("port", flaskServerPort());
+    Gson gson = new Gson();
+    String contextAsString = gson.toJson(context);
+    return new AutotranslationServiceImpl(contextAsString);
+  }
+
+  private AutotranslationServiceImpl(String configuration) {
+    Gson gson = new Gson();
+    Map map = gson.fromJson(configuration, Map.class);
+    String c = (String) map.get("contextId");
+    String port = (String) map.get("port");
+    this.context = new AutotranslationContext(c, port);
+    this.contextAsString = configuration;
+  }
+
+  public String getContextAsString() {
+    return contextAsString;
+  }
+
+  @Override
+  public String update(String name, String json) {
+    try {
+      String reply = Request.Post(LOCALHOST + this.context.getPort() + AUTOTRANSLTION)
+              .addHeader(AUTHORIZATION, auth())
+              .bodyString(createBody(name, json), ContentType.APPLICATION_JSON)
+              .execute().returnContent().asString();
+      if (!reply.equals("ok")) {
+        throw new RuntimeException(reply);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+    return json;
+  }
+
+  @Override
+  public String get(String name) {
+    String valueString = "";
+    try {
+      valueString = Request
+              .Get(LOCALHOST + this.context.getPort() + AUTOTRANSLTION + this.context.getContextId() + "/" + name)
+              .addHeader(AUTHORIZATION, auth())
+              .execute()
+              .returnContent()
+              .asString();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    return valueString;
+  }
+
+  private String createBody(String name, String json) {
+    Map<String, String> context = new HashMap<>();
+    context.put("name", name);
+    context.put("json", json);
+    context.put("sessionId", this.context.getContextId());
+    Gson gson = new Gson();
+    return gson.toJson(context);
+  }
+
+  private String auth() {
+    String authString = getBasic_auth_username() + ":" + getBasic_auth_password();
+    return "Basic " + Base64.encodeBase64String(authString.getBytes(StandardCharsets.UTF_8));
+  }
+
+  private static String getBasic_auth_username() {
+    return System.getenv("BEAKERX_BASIC_AUTH_USERNAME");
+  }
+
+  private static String getBasic_auth_password() {
+    return System.getenv("BEAKERX_BASIC_AUTH_PASSWORD");
+  }
+
+  private static String flaskServerPort() {
```

Comment:
rename flaskServerPort to autotranslationPort

Reply:
done